include/kodlab_mjbots_sdk/rotations.h

```diff
@@ -0,0 +1,590 @@
+/**
+ * @file rotations.h
+ * @author Ethan Musser (emusser@seas.upenn.edu)
+ * @brief 
+ * @version 
+ * @date 7/8/22
+ * 
+ * @copyright 2022 The Trustees of the University of Pennsylvania. All rights reserved.
+ * 
+ */
+
+#pragma once
+
+#include <Eigen/Geometry>
+#include <type_traits>
+#include <array>
+#include <algorithm>
+
+namespace kodlab::rotations
+{
+
+///////////////////////////////////////////////////////////////////////////////
+// Euler Angles Representation                                               //
+///////////////////////////////////////////////////////////////////////////////
+
+enum EulerRotationType
+{
+  EXTRINSIC = 0,
+  INTRINSIC
+};
+
+const std::array<int, 3> DEFAULT_EULER_ANGLE_AXES = {0, 1, 2};
+const EulerRotationType DEFAULT_EULER_ANGLE_ROTATION_TYPE = EXTRINSIC;
+
+/**
+ * @brief Euler angle rotation object.
+ * @note Default convention is <b>X-Y-Z Extrinsic</b> Euler angles.
+ *
+ * @tparam Scalar[optional] numeric type
+ */
+template<typename Scalar = float>
```

Comment:
My concern with this block of code is that if we are not using this right now and we have not tested it, it could easily be bugged.

Reply:
All untested portions have been commented as of 1c333dee37947609c1b04aec046276f38b3347ed.  Marking as resolved.